llvm/lib/ObjectYAML/GOFFEmitter.cpp

```diff
@@ -259,6 +313,14 @@ bool GOFFState::writeObject() {
   writeHeader(Doc.Header);
   if (HasError)
     return false;
+  // Iterate over all records.
+  for (auto &Rec : Doc.Records) {
```

Comment:
As the type of `Records` isn't immediately obvious, I think you shouldn't use `auto` here. Also, should it be `const`? Same goes for the `Sym` type below (`auto` is okay for `Sym` but `const should be added).